storage/src/main/java/io/delta/storage/commit/uccommitcoordinator/UCCommitCoordinatorClient.java

```diff
@@ -0,0 +1,912 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.storage.commit.uccommitcoordinator;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+import io.delta.storage.CloseableIterator;
+import io.delta.storage.LogStore;
+import io.delta.storage.commit.*;
+import io.delta.storage.commit.Commit;
+import io.delta.storage.commit.CommitCoordinatorClient;
+import io.delta.storage.commit.CommitFailedException;
+import io.delta.storage.commit.CoordinatedCommitsUtils;
+import io.delta.storage.commit.GetCommitsResponse;
+import io.delta.storage.commit.actions.AbstractMetadata;
+import io.delta.storage.commit.actions.AbstractProtocol;
+import io.delta.storage.internal.FileNameUtils;
+import io.delta.storage.internal.LogStoreErrors;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A commit coordinator client that uses unity-catalog as the commit coordinator.
+ */
+public class UCCommitCoordinatorClient implements CommitCoordinatorClient {
+  public UCCommitCoordinatorClient(Map<String, String> conf, UCClient ucClient) {
+    this.conf = conf;
+    this.ucClient = ucClient;
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(UCCommitCoordinatorClient.class);
+
+  private static final int SUPPORTED_READ_VERSION = 0;
+
+  private static final int SUPPORTED_WRITE_VERSION = 0;
+
+  private static final String READ_VERSION_KEY = "readVersion";
+
+  private static final String WRITE_VERSION_KEY = "writeVersion";
+
+  public final UCClient ucClient;
+
+  public final Map<String, String> conf;
+
+  final static public String UC_TABLE_ID_KEY = "ucTableId";
+
+  final static public String UC_METASTORE_ID_KEY = "ucMetastoreId";
```

Comment:
Can we add code comment for each of these variables?

Also we can keep all static together and move the non-static one together.